urllib3/response.py

```diff
@@ -150,7 +150,7 @@ def get_redirect_location(self):
         return False
 
     def release_conn(self):
-        if not self._pool or not self._connection:
+        if not self._pool or self._connection is None:
```

Comment:
My arguments would be:

1. We should be consistent, and we rely on falsey values everywhere else where None does not have a special meaning separate from False.
2. This bug happened precisely because we're doing a check for `is None` rather than a falsey check: https://github.com/shazow/urllib3/blob/master/urllib3/response.py#L255

I'd prefer we fix https://github.com/shazow/urllib3/blob/master/urllib3/response.py#L255 rather than change this to a None check.